src/main/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueue.java

```diff
@@ -1110,7 +1114,9 @@ private synchronized void headerRecovery(final SingleChronicleQueue that, final
                     }
                 } finally {
                     try {
-                        fileLock.release();
+                        if (fileLock != null) {
```

Comment:
won't it always be non null?

Reply:
No the earlier null check is inside the try block. I should move it out though.